aquadoggo_cli/src/config.rs

```diff
@@ -0,0 +1,374 @@
+#[derive(Parser, Serialize, Debug)]
+#[command(
+    name = "aquadoggo Node",
+    about = "Node server for the p2panda network",
+    version
+)]
+struct Cli {
+    /// Path to a config.toml file.
+    #[arg(short = 'c', long, value_name = "PATH")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    config: Option<PathBuf>,
+
+    /// List of schema ids which a node will replicate and expose on the GraphQL API.
+    #[arg(short = 's', long, value_name = "SCHEMA_ID SCHEMA_ID, ...", num_args = 0..)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    supported_schema_ids: Option<Vec<SchemaId>>,
+
+    /// URL / connection string to PostgreSQL or SQLite database.
+    #[arg(short = 'd', long, value_name = "CONNECTION_STRING")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    database_url: Option<String>,
+
+    /// HTTP port for client-node communication, serving the GraphQL API.
+    #[arg(short = 'p', long, value_name = "PORT")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    http_port: Option<u16>,
+
+    /// QUIC port for node-node communication and data replication.
+    #[arg(short = 'q', long, value_name = "PORT")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    quic_port: Option<u16>,
+
+    /// Path to persist your ed25519 private key file.
+    #[arg(short = 'k', long, value_name = "PATH")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    private_key: Option<PathBuf>,
+
+    /// mDNS to discover other peers on the local network.
+    #[arg(short = 'm', long, value_name = "BOOL")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    mdns: Option<bool>,
+
+    /// List of known node addresses we want to connect to directly.
+    #[arg(short = 'n', long, value_name = "IP:PORT IP:PORT, ...", num_args = 0..)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    direct_node_addresses: Option<Vec<SocketAddr>>,
+
+    /// Address of relay.
+    #[arg(short = 'r', long, value_name = "IP:PORT")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    relay_address: Option<SocketAddr>,
+
+    /// Set to true if our node should also function as a relay.
+    #[arg(short = 'e', long, value_name = "BOOL")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    relay_mode: Option<bool>,
+}
+
+/// Configuration derived from environment variables and .toml file.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Configuration {
+    pub supported_schema_ids: UncheckedAllowList,
+    pub database_url: String,
+    pub database_max_connections: u32,
+    pub worker_pool_size: u32,
+    pub http_port: u16,
+    pub quic_port: u16,
+    pub private_key: Option<PathBuf>,
+    pub mdns: bool,
+    pub direct_node_addresses: Vec<SocketAddr>,
+    pub relay_address: Option<SocketAddr>,
+    pub relay_mode: bool,
+}
+
+impl Default for Configuration {
+    fn default() -> Self {
+        Self {
+            supported_schema_ids: UncheckedAllowList::Set(vec![]),
```

Comment:
Been playing around with the cli (it's super fun) and I don't think it's possible to set `Wildcard` mode for supported_schema_ids.

Reply:
This would be quite useful, I'm looking into it now, let me know if i'm wrong in my understanding :+1:

Reply:
We could change the default to be `Wildcard` then all configurations will be available from the cli (only)